workflows/pkg/apis/workflows/v1alpha1/workflow_types.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2021 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+type Workflow struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec holds the desired state of the Workflow from the client
+	// +optional
+	Spec WorkflowSpec `json:"spec,omitempty"`
+
+	// +optional
+	Status WorkflowStatus `json:"status,omitempty"`
+}
+
+type WorkflowSpec struct {
+
+}
+
+
+type WorkflowStatus struct {
```

Comment:
might make sense to include some docstrings for these new structs to start off on a good foot? (of course we can add them later but sometimes later means ... much later XD)

Reply:
good point. added.